src/main/java/com/ibm/watson/litelinks/client/TServiceClientManager.java

```diff
@@ -297,11 +298,13 @@ public void onRemoval(RemovalNotification<ServiceKey, TServiceClientManager<?>>
                 pc = si.borrowClient();
             } catch (IllegalStateException ise) {
                 continue; // service inactive
-            } catch (TTransportException | SocketException te) {
+            } catch (TTransportException | SocketException | NoSuchElementException te) {
```

Comment:
Is there any exception that could be caught here that we wouldn't want to retry on?

Reply:
@tjohnson31415 that's a good question. Probably `InterruptedException` would be the only one, I'll update it.